core/src/window_service.rs

```diff
@@ -81,28 +81,49 @@ where
     let now = Instant::now();
```

Comment:
Would be nice to get rid of the `packets.packets.append` at some point. I don't think we need to copy into a single vec do we?